security/pkg/nodeagent/caclient/providers/citadel/client.go

```diff
@@ -119,12 +121,14 @@ func (c *citadelClient) getTLSDialOption() (grpc.DialOption, error) {
 		if err != nil {
 			return nil, err
 		}
+		citadelClientLog.Warna("Citadel client using public DNS: ", c.caEndpoint)
 	} else {
 		certPool = x509.NewCertPool()
 		ok := certPool.AppendCertsFromPEM(c.caTLSRootCert)
 		if !ok {
 			return nil, fmt.Errorf("failed to append certificates")
 		}
+		citadelClientLog.Infof("Citadel client using custom root: ", c.caEndpoint, " ", string(c.caTLSRootCert))
```

Comment:
Infof -> Debugf

Reply:
It is too hard to debug CA problems without this info.